### ISSUE TITLE
Weather APIs

### ISSUE THREAD

**Issue by Sibyx** [Fatra-Studios/froggo#1]
Research of available APIs to harvest data. Find out which data are available and which states of weather are possible to read.

**Comment by Sibyx** [comment]
Here is my analysis of [OpenWeatherMap.com](https://openweathermap.org/) service.

## TL;DR

We are able to get information according to the GPS coordinates. There is an `weather` array in API response, where we can find everything we need in string descriptors.

More information about available groups: [Weather condition codes](https://openweathermap.org/weather-conditions#Weather-Condition-Codes-2)

## Input

Firstly, I have to create account to have an access to the service. 

API is able to receive these type of input:
- City & country code
- City ID according to [their own database](http://bulk.openweathermap.org/sample/)
- GPS coordinates (it's probably our best choice)
- ZIP & country code

## Example request

Here is main example request for Martin:

```
GET https://api.openweathermap.org/data/2.5/weather
   ?lat=49.0822957
   &lon=18.929501
   &appid={API_KEY}
```

```json
{
  "coord": {
    "lon": 18.93,
    "lat": 49.08
  },
  "weather": [
    {
      "id": 600,
      "main": "Snow",
      "description": "light snow",
      "icon": "13n"
    }
  ],
  "base": "stations",
  "main": {
    "temp": 273.68,
    "pressure": 1025,
    "humidity": 90,
    "temp_min": 273.15,
    "temp_max": 274.15
  },
  "wind": {
    "speed": 0.5
  },
  "clouds": {
    "all": 90
  },
  "dt": 1545759000,
  "sys": {
    "message": 0.0021,
    "country": "SK",
    "sunrise": 1545719870,
    "sunset": 1545749493,
    "type": 1,
    "id": 7057
  },
  "id": 3058780,
  "name": "Martin",
  "cod": 200,
  "visibility": 8000
}
```

## Fields

Here is the list of interesting information from API response

- `weather`: array of weather description objects (THIS IS WHAT ARE WE LOOKING FOR)
   - `id`: an identification code from docs
   - `main`: group name (`Thunderstorm` | `Drizzle` | `Rain` | `Snow` | `Atmosphere` | `Clear` | `Clouds`)
   - `description`: [Weather condition codes](https://openweathermap.org/weather-conditions#Weather-Condition-Codes-2)
- `name`: Name of the city or location
- `clouds.all`: Cloudiness in %
- `main`: Object containing all available weather measures
